### PR TITLE
Request to merge working-branch into development branch

### PR DESCRIPTION
**Pull request by Haywhizzz**
Following the completion of the project, the authors request the merge from the working-branch to the development branch.

The following was implemented in the project:

- Users can create an account with their just a username

- Users can create events

- Users can see past events and upcoming ones

- Users can see all events created

- Users can register for events

- Test cases are passing

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def show
+    @user = User.find(params[:id])
+    @past = @user.attended_events.past
+    @upcoming = @user.attended_events.upcoming
+  end
```
Comment (lekegitrepo) [inline app/controllers/users_controller.rb]: - **_There should be a lists of events created by the user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))_**

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Request changes :recycle:
### Hi @Haywhizzz and @Abidoyinsola1 :clap: :clap:
#### Good job so far :+1: However, you need to consider the following :point_down:  issue before the project can be approved

- **_The project could not build as indicated in the image :point_down: 
Screenshot from 2020-10-19 10-44-56_**
**_So, this means that I could not run the migration command for the project, hence I was unable to run the project and the specs_**

#### _Reviewed by Temitayo Adeleke_

File: db/migrate/20201014101640_create_events.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class CreateEvents < ActiveRecord::Migration[6.0]
+  def change
+    create_table :events do |t|
+      t.string :name
+      t.text :description
+      t.string :location
+      t.datetime :date
+      t.timestamps
+      t.integer :creator_id
```
Comment (kubilaycaglayan) [inline db/migrate/20201014101640_create_events.rb:9]: :bulb:  events table has creator_id as the foreign key ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_assoc_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

**Review by kubilaycaglayan — CHANGES_REQUESTED** [review]
### Hello Team 👋
## Status: Changes Required ♻️

Great job so far! 👏👋
Thanks for NOT having any logic in the views 👏😊
Let's cover a few details together to make your project better ⏬

### GitFlow

- [x] ⚠️ Set your development branch as the default branch, please 😊

### Migrations
- [x] events table has creator_id as the foreign key ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_assoc_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

💡 You should make it Foreign Key explicitly. Currently, it is a column with the Integer data type.

There are a few ways to add foreign keys for association in migrations. You can:
- use `t.references` with index option https://api.rubyonrails.org/v5.1.7/classes/ActiveRecord/ConnectionAdapters/TableDefinition.html#method-i-references
- use `t.belongs_to` (alias for `references`) with index option https://api.rubyonrails.org/v5.1.7/classes/ActiveRecord/ConnectionAdapters/TableDefinition.html#method-i-belongs_to

- [x] We also need to Foreign Key to the `registrations` for `attendee_id` as User. Right? Please fix this too 🙏

Why indices for foreign keys are necessary? In short:
Since the foreign key columns will be used often to retrieve related information, it's important to index them (queries on those columns will be faster with indexes).
More details:
- https://semaphoreci.com/blog/2017/05/09/faster-rails-is-your-database-properly-indexed.html
- https://sqlperformance.com/2012/11/t-sql-queries/benefits-indexing-foreign-keys.

### Controllers
##### EventsController
###### `create` action
- [x] uses #build method on association, i.e. `current_user.events.build(event_params)` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/build_method_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

💡 A shortcut method what you are actually doing there. Try to understand it's benefits. 😊

##### UsersController
###### `show` action
- [x] lists events created by user, i.e. `@created_events =  current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

💡 I see that you can get this by `@user.events.each` in the view. I would be happy if you can follow the requirements 😊

### Tests
💡 Let's spice up the tests! 😊🙏
Currently, you only have 6 test cases.

- [x] models have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
- [x] authentication have has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
- [x] event managment features have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))

## That's it, you can request another code review request after making the changes ♻️😇

With all my respect 🙏
> Kubilay Caglayan | Technical Support Engineer

**Review by rOluochKe — APPROVED** [review]
Hello Team
### You got it right :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy coding!

**Pull request merged**